### ISSUE TITLE
Clarification of Distillation and Contrastive Learning

### ISSUE THREAD

**Issue by joubbi27** [stau-7001/S3AI#1]
Hi, I want to express my appreciation for your work. I would like to explore more the use of distillation and contrastive learning for the antibody structure and sequence.

In the paper on bioRxiv, Figure 2.a clearly shocases how the '_structural information distillation_'  training is performed and is further analyzed in section 4.3 presenting also the loss function used during the student training.
Nevertheless, I could not track the same process inside your codebase, maybe it is missing?

For example in '[common/losses.py](https://github.com/stau-7001/S3AI/blob/main/commons/losses.py)' the _NT-Xent loss function_ is not defined, and can be found in an [artifact of the same file](https://github.com/stau-7001/S3AI/blob/main/commons/.ipynb_checkpoints/losses-checkpoint.py#L245) in the .ipynb_checkpoints folder.
Another point that confuses me when looking at the example for sarscov2 data is the differences between the paper and the code. As reported in the README and config file it loads as model type the '_MultiTask_S3AI_Chem_' (https://github.com/stau-7001/S3AI/blob/main/configs/train_on_sarscov2.yml#L43).
Comparing the text in section 4.2 and the actual implementation (https://github.com/stau-7001/S3AI/blob/main/models/s3ai.py#L424), z_{ab}^{seq2str} should be computed using the 'structure encoder' which was pre-trained using the process mentioned above, while in the code is obtained by a forward pass of a MLP (https://github.com/stau-7001/S3AI/blob/main/models/s3ai.py#L476). 

Am I missing something, without the loss being defined anywhere in the code it seems like the distillation process is missing.

**Comment by stau-7001** [comment]
Thank you for your detailed feedback and for your interest in our work! We truly appreciate your thorough exploration of the paper and the codebase.
You are correct that the NT-Xent loss function and some aspects of the structural information distillation process described in the paper are not explicitly present in the current codebase. This is because the code related to the distillation process and certain loss functions has not yet been released. The current version of the code focuses on downstream task of the model.
At this stage, we are not planning to release the distillation-related code publicly just yet, as the model is still being refined. However, if you require access to this part of the code for your research or collaboration, please feel free to reach out to me directly via email. I would be happy to assist you and provide the necessary details.
